File: Dynamic Programming/Java/2748 피보나치 수2.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      long[] dp = new long[91];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 1;
      for(int i = 3; i <= n; i++) {
          dp[i] = dp[i-2] + dp[i-1];
      }
      
      System.out.format("%d", dp[n]);
    }
}

// http://boj.kr/1a92f899943249f5aaa47e1e9010b2c8

File: _etc/Kotlin/2748 피보나치 수2.kt
import java.util.Scanner

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val n = sc.nextInt()
    var dp = LongArray(91){0}
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 1
    
    for(i in 3..n) {
        dp[i] = dp[i-1] + dp[i-2]
    }
    
    println(dp[n])
}

// http://boj.kr/a006d69268ff4b0ca1d9031641e09659
